storage/replica.go

```diff
@@ -2111,6 +2111,13 @@ func (r *Replica) loadSystemConfigSpan() ([]roachpb.KeyValue, []byte, error) {
 	return kvs, config.SystemConfig{Values: kvs}.Hash(), nil
 }
 
+// needsSplitBySize returns true if the size of the range requires it
+// to be split.
+func (r *Replica) needsSplitBySize() bool {
+	maxBytes := r.GetMaxBytes()
+	return maxBytes > 0 && r.stats.KeyBytes+r.stats.ValBytes > maxBytes
```

Comment:
It is, that's what failed in CI.

On Mon, May 2, 2016 at 4:11 PM, Tobias Schottdorf <notifications@github.com>
wrote:

> In storage/replica.go
> <https://github.com/cockroachdb/cockroach/pull/6447#discussion_r61795681>:
>
> > @@ -2111,6 +2111,13 @@ func (r *Replica) loadSystemConfigSpan() ([]roachpb.KeyValue, []byte, error) {
> >  	return kvs, config.SystemConfig{Values: kvs}.Hash(), nil
> >  }
> >
> > +// needsSplitBySize returns true if the size of the range requires it
> > +// to be split.
> > +func (r *Replica) needsSplitBySize() bool {
> > +	maxBytes := r.GetMaxBytes()
> > +	return maxBytes > 0 && r.stats.KeyBytes+r.stats.ValBytes > maxBytes
>
> how is accessing r.stats not racy?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/6447/files/89b63187315656c641c369176a87e427f22a5f32#r61795681>
>